sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/aio/_azureappconfigurationproviderasync.py

```diff
@@ -226,11 +221,113 @@ class AzureAppConfigurationProvider(Mapping[str, Union[str, JSON]]):
     keys. Enables resolution of Key Vault references in configuration settings.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, **kwargs) -> None:
         self._dict: Dict[str, str] = {}
         self._trim_prefixes: List[str] = []
         self._client: Optional[AzureAppConfigurationClient] = None
         self._secret_clients: Dict[str, SecretClient] = {}
+        self._key_vault_options: Optional[AzureAppConfigurationKeyVaultOptions] = kwargs.pop("key_vault_options", None)
+        self._selects: List[SettingSelector] = kwargs.pop(
+            "selects", [SettingSelector(key_filter="*", label_filter=EMPTY_LABEL)]
+        )
+
+        trim_prefixes: List[str] = kwargs.pop("trim_prefixes", [])
+        self._trim_prefixes = sorted(trim_prefixes, key=len, reverse=True)
+
+        refresh_on: List[Tuple[str, str]] = kwargs.pop("refresh_on", None) or []
+        self._refresh_on: Mapping[Tuple[str, str] : Optional[str]] = {_build_sentinel(s): None for s in refresh_on}
+        self._refresh_timer: _RefreshTimer = _RefreshTimer(**kwargs)
+        self._on_refresh_error: Optional[Callable[[Exception], None]] = kwargs.pop("on_refresh_error", None)
+        self._update_lock = Lock()
+
+    async def refresh(self, **kwargs) -> None:
+        if not self._refresh_on:
+            logging.debug("Refresh called but no refresh options set.")
+            return
+
+        if not self._refresh_timer.needs_refresh():
+            logging.debug("Refresh called but refresh interval not elapsed.")
+            return
+        try:
+            with self._update_lock:
+                for (key, label), etag in self._refresh_on.items():
+                    updated_sentinel = await self._client.get_configuration_setting(
+                        key=key, label=label, etag=etag, match_condition=MatchConditions.IfModified, **kwargs
+                    )
+                    if updated_sentinel is not None:
+                        logging.debug(
+                            "Refresh all triggered by key: %s label %s.",
+                            key,
+                            label,
+                        )
+                        await self._load_all(**kwargs)
+                        self._refresh_on[(key, label)] = updated_sentinel.etag
+                        self._refresh_timer.reset()
+                        return
+        except (ServiceRequestError, ServiceResponseError) as e:
+            logging.debug("Failed to refresh, retrying: %r", e)
+            self._refresh_timer.retry()
+        except HttpResponseError as e:
+            # If we get an error we should retry sooner than the next refresh interval
+            self._refresh_timer.retry()
+            if _is_retryable_error(e):
+                return
+            if self._on_refresh_error:
+                self._on_refresh_error(e)
```

Comment:
We should support asynchronous callbacks for the async implementation - so this will need an `await`.